duffy/database/model/tenant.py

```diff
@@ -10,4 +14,20 @@ class Tenant(Base, CreatableMixin, RetirableMixin):
     id = Column(Integer, primary_key=True, nullable=False)
     name = Column(UnicodeText, unique=True, nullable=False)
     is_admin = Column(Boolean, nullable=False, default=False, server_default=text("FALSE"))
+    api_key = Column(Text, nullable=False)
     ssh_key = Column(UnicodeText, nullable=False)
+    _api_key = Column("api_key", Text, nullable=False)
+
+    @hybrid_property
+    def api_key(self):
+        return self._api_key
+
+    @api_key.setter
+    def api_key(self, key: uuid.UUID):
+        salt = bcrypt.gensalt()
+        hashed_key = bcrypt.hashpw(str(key).encode("ascii"), salt)
+        self._api_key = hashed_key.decode("ascii")
+
+
+def validate_api_key(key: str, hash: str):
+    return bcrypt.checkpw(str(key).encode("ascii"), hash.encode("ascii"))
```

Comment:
I'd rather have this as a method of `Tenant` objects (with its own test):

```
    def api_key(...):
        ...

    def validate_api_key(self, key: uuid.UUID) -> bool:
        return bcrypt.checkpw(str(key).encode("ascii"), self.api_key.encode("ascii"))
```

Would be used somehow like this:

```
    result = await DBSession.execute(select(Tenant).filter_by(...))
    tenant = result.scalar_one_or_none()
    if not (tenant and tenant.validate_api_key(key)):
        raise HTTPException(...)
```

Reply:
I have restored the `validate_api_key` (previously) function as a method under the `Tenant` class but I did not add the latter parts as they are more involved in the authentication which is beyond the scope of this pull request. I'd address those when I'll get to work on a PR to fix #135.

Reply:
Ok, scratch whatever I said at https://github.com/CentOS/duffy/pull/146#discussion_r770183711. I would need to implement the latter parts as well so as to write the tests for the `Tenant` object - even though the PR is not about it :/